### ISSUE TITLE
spring-data-rest deserialization support 

### ISSUE THREAD

**Issue by EugenMayer** [Blazebit/blaze-persistence#921]
Testing the new spring-data-rest deserialization i created a PoC

https://github.com/EugenMayer/java-spring-pocs/blob/master/src/main/java/de/kontextwork/poc/spring/blaze/fullapp/subject/group/controller/GroupController.java

With the test here

https://github.com/EugenMayer/java-spring-pocs/blob/master/src/test/java/de/kontextwork/poc/spring/blaze/fullapp/subject/group/controller/GroupControllerTest.java#L107

I am running against the current master snapshot for now.

Steps i took to integrate:

- add "com.blazebit:blaze-persistence-integration-spring-data-rest:${blazePersistence}"

When trying to run the tests i get an exception about the sorter 

```
Caused by: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'sortResolver' defined in class path resource [org/springframework/data/rest/webmvc/config/RepositoryRestMvcConfiguration.class]: Cannot register bean definition [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.springframework.data.rest.webmvc.config.RepositoryRestMvcConfiguration; factoryMethodName=sortResolver; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/springframework/data/rest/webmvc/config/RepositoryRestMvcConfiguration.class]] for bean 'sortResolver': There is already [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=com.blazebit.persistence.spring.data.rest.impl.BlazePersistenceWebConfiguration; factoryMethodName=sortResolver; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/blazebit/persistence/spring/data/rest/impl/BlazePersistenceWebConfiguration.class]] bound.
```

I guess this comes from `BlazePersistenceWebConfiguration` being auto exposed and auto configured.
Considering that i do not define my own sorter - i do actually not override anything here:
https://github.com/EugenMayer/java-spring-pocs/blob/master/src/main/java/de/kontextwork/poc/spring/configuration/RestConfiguration.java#L23

I assume this exception would be thrown on every single application - not just this poc. I think we are trying to auto-reconfigure the spring default which is defined in `SpringDataWebConfiguration`

and is actually already the same. In this case either we are doing the integration the wrong way or in this specific case i would tend to go with you should not define this bean at all since it is the default and the spring one also adds `customizeSortResolver(sortResolver);` which is probably a different hook to customize the sorter in general - in any way we are defining a `@Bean` which is already the default -> IMHO remove it

**Comment by EugenMayer** [comment]
#922 fixes this issue and actually also the whole thing seems to be working know - deserialization works in this PoC setup. Great work there @beikov

**Comment by beikov** [comment]
Merged via d7fef8942f16d325c22a71fd9bd14242e0954834

**Comment by Mobe91** [comment]
@EugenMayer 
This breaks auto configuration of blaze-persistence-integration-spring-data-webmvc when spring-data-rest is not used. In this case `org.springframework.boot.autoconfigure.data.web.SpringDataWebAutoConfiguration` is not applied since it is marked with `@ConditionalOnMissingBean(PageableHandlerMethodArgumentResolver.class)` ->  `org.springframework.data.web.config.SpringDataWebConfiguration` is not imported -> no bean for `org.springframework.data.web.SortHandlerMethodArgumentResolver` is defined.

Maybe we should provide a separate `SortHandlerMethodArgumentResolverConfiguration` that contains an 
```java
@ConditionalOnMissingBean
@Bean
public SortHandlerMethodArgumentResolver sortResolver() {
   // also consider SortHandlerMethodArgumentResolverCustomizer
}
```
This should be imported from `com.blazebit.persistence.spring.data.webmvc.impl.BlazePersistenceWebConfiguration`. Not 100% sure if it works like that, though.
